### PR TITLE
batchupload: tag new allocated records as migrated

### PR DESCRIPTION
**Pull request by zzacharo**

### CODE REVIEW

File: modules/bibupload/lib/batchuploader_webinterface.py [hunk 1]
```diff
@@ -184,6 +191,36 @@ def allocaterecord(self, req, form):
...
+                                        <subfield code="c">MIGRATED</subfield>
+                                    </datafield>
+                               </record> """ % {
+                'recid':
+                recid,
+                'useragent':user_agent
+            }
+            tmp.write(marc_content)
+            tmp.seek(0)
+            task_arguments = ('bibupload', 'admin', '-r', path,
```
Comment (ludmilamarian) [inline modules/bibupload/lib/batchuploader_webinterface.py]: why priority -3?

File: modules/bibupload/lib/batchuploader_webinterface.py [hunk 2]
```diff
@@ -184,6 +191,36 @@ def allocaterecord(self, req, form):
...
+        """
+        recid = cli_allocate_record(req)
+        user_info = collect_user_info(req)
+        user_agent = user_info['agent']
+        fd, path = tempfile.mkstemp(suffix='.xml', dir=CFG_TMPDIR)
+        with os.fdopen(fd, 'w') as tmp:
+            marc_content = """ <record>
+                                    <controlfield tag="001">%(recid)s</controlfield>
+                                    <datafield tag="970" ind1=" " ind2=" ">
+                                        <subfield code="d">https://%(useragent)s.cern.ch/record/%(recid)s</subfield>
```
Comment (ludmilamarian) [inline modules/bibupload/lib/batchuploader_webinterface.py]: we should validate `user_agent` against a known list (allowed values.. to make sure we don't put strange stuff in 970). Also, do we need to change the config somewhere to allow more user agents? AFAIR the only allowed user agent was `invenio`.

File: modules/bibupload/lib/batchuploader_webinterface.py [hunk 3]
```diff
@@ -184,6 +191,36 @@ def allocaterecord(self, req, form):
         """
         return cli_allocate_record(req)
 
+    def allocaterecordandtagasmigrated(self, req, form):
```
Comment (ludmilamarian) [inline modules/bibupload/lib/batchuploader_webinterface.py]: can we call it `allocatemigratedrecord` to be slightly shorter?

File: modules/bibupload/lib/batchuploader_webinterface.py [hunk 4]
```diff
@@ -48,8 +53,10 @@
 class WebInterfaceBatchUploaderPages(WebInterfaceDirectory):
     """Defines the set of /batchuploader pages."""
 
-    _exports = ['', 'metadata', 'metasubmit', 'history', 'documents',
-        'docsubmit', 'daemon', 'allocaterecord', 'confirm']
+    _exports = [
+        '', 'metadata', 'metasubmit', 'history', 'documents', 'docsubmit',
+        'daemon', 'allocaterecord', 'allocaterecordandtagasmigrated', 'confirm'
```
Comment (ludmilamarian) [inline modules/bibupload/lib/batchuploader_webinterface.py]: you would need to change the name here as well

File: modules/bibupload/lib/batchuploader_webinterface.py [hunk 5]
```diff
@@ -184,6 +191,34 @@ def allocaterecord(self, req, form):
         """
         return cli_allocate_record(req)
 
+    def allocatemigratedrecord(self, req, form):
+        """
+        Interface for allocating a record and tag it as migrated in order to redirect
+        to `videos` satellite.
```
Comment (ludmilamarian) [inline modules/bibupload/lib/batchuploader_webinterface.py]: replace `videos` with `a CDS satellite, for ex. videos`

File: modules/bibupload/lib/batchuploader_webinterface.py (hunk 4 above)
Comment (zzacharo) [reply modules/bibupload/lib/batchuploader_webinterface.py]: should I remove the `allocaterecord ` or not?

Comment (ludmilamarian) [reply modules/bibupload/lib/batchuploader_webinterface.py]: nop, I think it is still used by other services.

**Review by ludmilamarian — APPROVED** [review]

**Pull request merged**